src/libswid.cpp

```diff
@@ -76,20 +76,27 @@ unsigned int swid_root_get_type(SWIDHandle data) {
  */
 
 SWIDIOHandle swid_create_io(const char * backend) {
-	return (SWIDIOHandle)get_swidtagio(backend);
+	SWIDIOHandle io = new SWIDTagIO();
+	try {
+		io->setBackend(backend);
+	} catch (...) {
+		return nullptr;
+	}
+
+	return io;
 }
 
 
 int swid_destroy_io(SWIDIOHandle io_handle) {
-	delete (SWIDTagIO *)io_handle;
+	delete (SWIDTagIOBase *)io_handle;
 	return 0;
 }
 
 
 int swid_load_root(SWIDIOHandle io_handle, const char * fname, SWIDHandle swid_structure) {
 	auto data = SWIDStruct();
 	try {
-		data = ((SWIDTagIO *)io_handle)->load(fname);
+		data = ((SWIDTagIOBase *)io_handle)->load(fname);
 	} catch (const XMLIOError & err) {
 		return 1;
```

Comment:
I would hate calling this with an invalid XML without it telling me what's wrong on which line..

Reply:
`swig_error_msg()` to get the last error string?

Reply:
I totally agree. However, I would like to leave this as it is for the time being. It is still an MVP, the C interface is not likely to be used right away, and this is about extending the interface, so those would be backwards-compatible changes.